config/filter.d/zoneminder.conf

```diff
@@ -0,0 +1,23 @@
+# Fail2Ban filter for Zoneminder login failures
+#
+
+[Definition]
+
+# pattern: [client 10.1.1.1:38022] WAR [Login denied for user "test"], referer: https://zoneminderurl/
+#
+#
+# Option:  failregex
+# Notes.:  regex to match the password failure messages in the logfile. The
+#          host must be matched by a group named "host". The tag "<HOST>" can
+#          be used for standard IP/hostname matching and is only an alias for
+#          (?:::f{4,6}:)?(?P<host>[\w\-.^_]+)
+
+
+failregex = ^.*? [[]client <HOST>:\d\d\d\d\d] WAR [[]Login denied for user \S*], referer: \S*$
```

Comment:
Because `.*?` is very "evil" at all (anchored or not). Please prevent usage of catch-all's (also if it would be greedy) if not really expected.
The benefit is:
- it will match exactly that (resp. precisely) and not any sub string inside log message.
- It finds matched entry or bypasses non-matched much faster (because the regular expressions with non-greedy catch-all's are slowest, and the bigger the message the slower it does searching);

So I'm with @yarikoptic and I will recommend following one:
```diff
-failregex = ^.*? [[]client <HOST>:\d\d\d\d\d] WAR [[]Login denied for user \S*], referer: \S*$
+failregex = "^\[\](?: \[:error\])?(?: \[pid \d+\])? \[client <HOST>:\d+\] WAR \[Login denied for user "[^"]*"\]"
```
I've removed referer from regexp, because it doesn't matter (possible miss if empty?)